code/fireball/fireballs.cpp

```diff
@@ -863,51 +837,58 @@ int fireball_create(vec3d *pos, int fireball_type, int render_type, int parent_o
 
 	obj = &Objects[objnum];
 
-	fb->fireball_info_index = fireball_type;
-	fb->fireball_render_type = render_type;
-	fb->time_elapsed = 0.0f;
-	fb->objnum = objnum;
-	fb->current_bitmap = -1;
+	new_fireball.fireball_info_index = fireball_type;
+	new_fireball.fireball_render_type = render_type;
+	new_fireball.time_elapsed = 0.0f;
+	new_fireball.objnum = objnum;
+	new_fireball.current_bitmap = -1;
 	
-	switch( fb->fireball_render_type )	{
+	switch( new_fireball.fireball_render_type )	{
 
 		case FIREBALL_MEDIUM_EXPLOSION:	
-			fb->orient = (myrand()>>8) & 7;							// 0 - 7
+			new_fireball.orient = (myrand()>>8) & 7;							// 0 - 7
 			break;
 
 		case FIREBALL_LARGE_EXPLOSION:
-			fb->orient = (myrand()>>8) % 360;						// 0 - 359
+			new_fireball.orient = (myrand()>>8) % 360;						// 0 - 359
 			break;
 
 		case FIREBALL_WARP_EFFECT:
 			// Play sound effect for warp hole opening up
-			fireball_play_warphole_open_sound(ship_class, fb);
+			fireball_play_warphole_open_sound(ship_class, &new_fireball);
 
 			// warp in type
 			if (reverse)	{
-				fb->orient = 1;
+				new_fireball.orient = 1;
 				// if warp out, then reverse the orientation
 				vm_vec_scale( &obj->orient.vec.fvec, -1.0f );	// Reverse the forward vector
 				vm_vec_scale( &obj->orient.vec.rvec, -1.0f );	// Reverse the right vector
 			} else {
-				fb->orient = 0;
+				new_fireball.orient = 0;
 			}
 			break;
 
 		default:
-			Int3();
+			UNREACHABLE("Bad type set in fireball_create");
 			break;
 	}
 
-	if ( fb->fireball_render_type == FIREBALL_WARP_EFFECT )	{
+	if ( new_fireball.fireball_render_type == FIREBALL_WARP_EFFECT )	{
 		Assert( warp_lifetime >= 4.0f );		// Warp lifetime must be at least 4 seconds!
 		if ( warp_lifetime < 4.0f )
 			warp_lifetime = 4.0f;
-		fb->total_time = warp_lifetime;	// in seconds
+		new_fireball.total_time = warp_lifetime;	// in seconds
 	} else {
-		fb->total_time = i2fl(fl->num_frames) / fl->fps;	// in seconds
+		new_fireball.total_time = i2fl(fl->num_frames) / fl->fps;	// in seconds
 	}
-	
+
+	if (n == static_cast<int>(Fireball_vec.size())) {
```

Comment:
If you keep the fireball a pointer and just point that to the correct element around line 787 then you can avoid having that very specific check here.